ipa-core/src/secret_sharing/array.rs

```diff
@@ -0,0 +1,422 @@
+use std::{
+    array,
+    borrow::Borrow,
+    fmt::Debug,
+    ops::{Add, AddAssign, Mul, Neg, Not, Sub, SubAssign},
+};
+
+use generic_array::{ArrayLength, GenericArray};
+use typenum::{U1, U32};
+
+use crate::{
+    ff::{Field, Fp32BitPrime, Serializable},
+    helpers::Message,
+    protocol::prss::{FromRandom, FromRandomU128},
+    secret_sharing::{FieldArray, SharedValue, SharedValueArray},
+};
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct StdArray<V: SharedValue, const N: usize>([V; N]);
```

Comment:
can you please document why we need to wrap a standard array? I assume because we need to implement some traits, but it would be great to have it written somewhere